### PR TITLE
#9 reset camera position using the "r" key

### PR DESCRIPTION
**Pull request by stefanrinderle**
See #9 for details.

### CODE REVIEW

File: softvis3d-frontend/src/components/scene/visualization/SoftVis3dScene.ts [hunk 1]
```diff
@@ -150,6 +149,15 @@ export default class SoftVis3dScene {
         return result;
     }
 
+    /**
+     * Why recalculate?
+     * I want the position to be reset based on the given shapes. In the use case of the "direct link" feature,
+     * reset to the "first" position would reset to the starting point from the link.
```
Comment (Ungolianth) [inline softvis3d-frontend/src/components/scene/visualization/SoftVis3dScene.ts]: Also the position will be correct after the shapes were changed and the objects were updated in the scene directly. 👍

File: softvis3d-frontend/test/components/scene/Scene.spec.tsx [hunk 2]
```diff
@@ -10,15 +10,47 @@ import SoftVis3dScene from "../../../src/components/scene/visualization/SoftVis3
 
 describe("<Scene/>", () => {
 
-    it("should initialize rendering", () => {
+    it("should initialize mounr", () => {
```
Comment (Ungolianth) [inline softvis3d-frontend/test/components/scene/Scene.spec.tsx]: mount?

File: softvis3d-frontend/test/components/scene/Scene.spec.tsx [hunk 3]
```diff
@@ -10,15 +10,47 @@ import SoftVis3dScene from "../../../src/components/scene/visualization/SoftVis3
...
+        localSceneStore.scenePainter = scenePainter;
+        let resetStub = Sinon.stub(scenePainter, "resetCameraPosition");
+
+        let underTest: Scene = new Scene();
+        underTest.props = {
+            sceneStore: localSceneStore
+        };
+
+        let event: any = {
+            keyCode: 82
```
Comment (Ungolianth) [inline softvis3d-frontend/test/components/scene/Scene.spec.tsx]: maybe use the static field in `Scene`

File: softvis3d-frontend/src/components/scene/Scene.tsx [hunk 4]
```diff
@@ -14,6 +14,9 @@ interface SceneProps {
 @observer
 export default class Scene extends React.Component<SceneProps, any> {
 
+    private static KEY_DOWN_EVENT_NAME: string = "keydown";
+    private static R_KEY_CODE: number = 82;
```
Comment (stefanrinderle) [inline softvis3d-frontend/src/components/scene/Scene.tsx]: Renamed to KEY_CODE_R.

File: softvis3d-frontend/test/components/scene/Scene.spec.tsx (hunk 3 above)
Comment (stefanrinderle) [inline softvis3d-frontend/test/components/scene/Scene.spec.tsx]: I did not use the static field in Scene intentionally. I want to assure that the "r" key is used for reset. The number 82 for the key r will not change. The test should be red if someone changes the number in Scene by mistake. Using the static variable would not find this.

**Comment by Ungolianth** [comment]
I did a few tests and everything seems fine. The only (and hopefully last) issue I have, is that the `keydown` event is registered to the `window`: We cannot in good conscience add an Input field, because the event will still trigger anytime I type a "R"... 
I assume any other SonarQube fields (eg. search) are also affected...

**Comment by stefanrinderle** [comment]
That's why we do the review ;-) Yes, attaching the event listener to the window has side effects. The event is also fired and processed if you search for a name with an "r" in the SQ search. I tried to attach the event listener to the document but this did not work. Do you  have an idea how we could limit the listener?

**Comment by stefanrinderle** [comment]
Close pull request because there is more work to do. We will open a new one once the work is done.

**Pull request closed without merging**